solidity/contracts/bancorx/BancorX.sol

```diff
@@ -384,7 +385,9 @@ contract BancorX is IBancorX, TokenHolder, ContractRegistryClient {
             // otherwise, verify transaction details
             require(txn.to == _to && txn.amount == _amount && txn.fromBlockchain == _fromBlockchain, "ERR_TX_MISMATCH");
 
-            if (_xTransferId != 0) require(transactionIds[_xTransferId] == _txId, "ERR_TX_ALREADY_EXISTS");
+            if (_xTransferId != 0) {
+                require(transactionIds[_xTransferId] == _txId, "ERR_TX_ALREADY_EXISTS");
```

Comment:
Can simplify this to:
```
require(_xTransferId == 0 || transactionIds[_xTransferId] == _txId, "ERR_TX_ALREADY_EXISTS");
```

Reply:
This PR is already too big, so let's postpone unrelated old code refactoring (unlike style or formatting)